docs/01-app/05-api-reference/05-config/01-next-config-js/taint.mdx

```diff
@@ -0,0 +1,237 @@
+---
+title: taint
+description: Enable tainting Objects and Values.
+version: experimental
+---
+
+## Usage
+
+Experimental support for tainting Objects and Values.
+
+This flag enables usage of these React features:
+
+- [`experimental_taintObjectReference`](https://react.dev/reference/react/experimental_taintObjectReference)
+- [`experimental_taintUniqueValue`](https://react.dev/reference/react/experimental_taintUniqueValue)
+
+> **Good to know**: Activating this flag, enables the React `experimental` channel for `app` directory
+
+```ts filename="next.config.ts" switcher
+import type { NextConfig } from 'next'
+
+const nextConfig: NextConfig = {
+  experimental: {
+    ppr: 'incremental',
+  },
+}
+
+export default nextConfig
+```
+
+```js filename="next.config.js" switcher
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  experimental: {
+    ppr: 'incremental',
+  },
+}
+
+module.exports = nextConfig
+```
+
+The taint API includes two experimental functions, that allow you to mark objects and values, to prevent them from ever reaching the Client.
+
+- To taint objects references, use `experimental_taintObjectReference`.
+- To taint unique values, use `experimental_taintUniqueValue`.
+
+> **Warning:** Do not rely on the taint API, as your only mechanism to prevent exposing sensitive data to the client.
```

Comment:
Let's link to Seb's blog post for now? 

```suggestion
> **Warning:** Do not rely on the taint API as your only mechanism to prevent exposing sensitive data to the client. See our [security recommendations](/blog/security-nextjs-server-components-actions).
```